react/templates/src/Profile.js

```diff
@@ -23,14 +23,15 @@ export default class Profile extends Component {
 
   render() {
     const { handleSignOut, userSession } = this.props;
+    const username = this.props.userSession.loadUserData().username;
     const { person } = this.state;
     return (
       !userSession.isSignInPending() ?
       <div className="panel-welcome" id="section-2">
         <div className="avatar-section">
           <img src={ person.avatarUrl() ? person.avatarUrl() : avatarFallbackImage } className="img-rounded avatar" id="avatar-image" alt=""/>
         </div>
-        <h1>Hello, <span id="heading-name">{ person.name() ? person.name() : 'Nameless Person' }</span>!</h1>
+        <h1>Hello, <span id="heading-name">{ person.name ? username : 'Nameless Person' }</span>!</h1>
```

Comment:
I think this should be:
```js
person.name() || username || 'Nameless Person'
```
Otherwise it will never show the human entered name

Reply:
It does with my change, at least with my tests.  I'll give your solution a try tho.